tools/autograd/templates/variable_factories.h

```diff
@@ -60,7 +61,7 @@ inline at::Tensor from_blob(
     const Deleter& deleter,
     const at::TensorOptions& options = at::TensorOptions()) {
   at::Tensor tensor = ([&]() {
-    at::AutoNonVariableTypeMode non_var_type_mode(true);
```

Comment:
@ezyang I deleted most of `AutoNonVariableTypeMode` according to your comments on #38502, but I didn't ask there: what's the reason we had `AutoNonVariableTypeMode` at these places in the first place? was it solely to skip tracing logic? would rest part of VariableType (e.g.: things inside `if (compute_requires_grad...)` branch) have any undesired side effect for other ops being called? Are we making decision based on the fact that at::from_blob / at::empty / etc won't involve autograd logic at all?

Reply:
Solely to skip tracing. This is something you have to work out by looking what gets called inside.

Actually, it was a bit *wrong* to skip variable mode, for example, I bet that `from_blob` when passed `requires_grad=True` didn't actually respect it. Your change here fixes that!

Reply:
Hmm, seems many CI failed because of removing AutoNonVariableTypeMode guard:
```
RuntimeError: Cannot insert a Tensor that requires grad as a constant. Consider making it a parameter or input, or detaching the gradient
```
I'll need figure out which one(s) of them are still necessary :'(